File: ALODWebUtility/Workflow/WorkOptionRule.vb
Imports ALOD.Core.Utils
Imports ALOD.Data

Namespace Worklfow

    Public Class WorkOptionRule
        Protected _id As Integer
        Protected _optionId As Integer
        Protected _ruleKey As String
        Protected _ruleType As RuleType
        Protected _ruleValue As String

        Public Sub New()
            _id = 0
            _optionId = 0
            _ruleKey = String.Empty
            _ruleValue = String.Empty
        End Sub

        Public Sub New(ByVal optionid As Integer, ByVal ruleKey As String, ByVal ruleVal As String, ByVal ruleType As RuleType)
            _optionId = optionid
            _ruleKey = ruleKey
            _ruleValue = ruleVal
            _ruleType = ruleType
        End Sub

        Public Property Id() As Integer
            Get
                Return _id
            End Get
            Set(ByVal value As Integer)
                _id = value
            End Set
        End Property

        Public Property OptionId() As Integer
            Get
                Return _optionId
            End Get
            Set(ByVal value As Integer)
                _optionId = value
            End Set
        End Property

        Public Property RuleKey() As String
            Get
                Return _ruleKey
            End Get
            Set(ByVal value As String)
                _ruleKey = value
            End Set
        End Property

        Public Property RuleTypeID() As RuleType
            Get
                Return _ruleType
            End Get
            Set(ByVal value As RuleType)
                _ruleType = value
            End Set
        End Property

        Public Property RuleValue() As String
            Get
                Return _ruleValue
            End Get
            Set(ByVal value As String)
                _ruleValue = value
            End Set
        End Property

        Public Sub Save()
            Dim adapter As New SqlDataStore()
            adapter.ExecuteNonQuery("core_workOptionRules_sp_Save", _optionId, _ruleKey, _ruleValue, _ruleType)
        End Sub

    End Class

End Namespace

File: ALODWebUtility/Workflow/WorkOptionRule.cs
using System;
using ALOD.Core.Utils;
using ALOD.Data;

namespace ALODWebUtility.Workflow
{
    public class WorkOptionRule
    {
        protected int _id;
        protected int _optionId;
        protected string _ruleKey;
        protected RuleType _ruleType;
        protected string _ruleValue;

        public WorkOptionRule()
        {
            _id = 0;
            _optionId = 0;
            _ruleKey = string.Empty;
            _ruleValue = string.Empty;
        }

        public WorkOptionRule(int optionid, string ruleKey, string ruleVal, RuleType ruleType)
        {
            _optionId = optionid;
            _ruleKey = ruleKey;
            _ruleValue = ruleVal;
            _ruleType = ruleType;
        }

        public int Id
        {
            get { return _id; }
            set { _id = value; }
        }

        public int OptionId
        {
            get { return _optionId; }
            set { _optionId = value; }
        }

        public string RuleKey
        {
            get { return _ruleKey; }
            set { _ruleKey = value; }
        }

        public RuleType RuleTypeID
        {
            get { return _ruleType; }
            set { _ruleType = value; }
        }

        public string RuleValue
        {
            get { return _ruleValue; }
            set { _ruleValue = value; }
        }

        public void Save()
        {
            SqlDataStore adapter = new SqlDataStore();
            adapter.ExecuteNonQuery("core_workOptionRules_sp_Save", _optionId, _ruleKey, _ruleValue, _ruleType);
        }
    }
}
